spec/routemaster/cache_spec.rb

```diff
@@ -4,112 +4,109 @@
-        make_response(@counter += 1)
-      end
-    end
+    describe '#get' do
+      let(:perform) { subject.get(url, **options) }
 
-    it 'fetches the url' do
-      expect(fetcher).to receive(:get).with(url, anything)
-      performer.call(url).status
-    end
+      context 'with no options' do
+        let(:perform) { subject.get(url) }
 
-    it 'passes the locale header' do
-      expect(fetcher).to receive(:get) do |url, **options|
-        expect(options[:headers]['Accept-Language']).to eq('fr')
-        make_response(1)
-      end
-      performer.call(url, locale: 'fr').status
-    end
+        it 'calls get on the fetcher with no version and locale headers' do
+          expect_any_instance_of(Fetcher)
+            .to receive(:get)
+            .with(url, headers: { 'Accept' => 'application/json' })
 
-    it 'passes the version header' do
-      expect(fetcher).to receive(:get) do |url, **options|
-        expect(options[:headers]['Accept']).to eq('application/json;v=33')
-        make_response(1)
+          perform
+        end
       end
-      performer.call(url, version: 33).status
-    end
 
-    it 'uses the cache' do
-      expect(fetcher).to receive(:get).once
-      3.times { performer.call(url).status }
-      expect(performer.call(url).body.t).to eq(1)
-    end
+      context 'with a specific version' do
+        let(:options) { { version: 2 } }
 
-    context 'with a listener' do
-      before { subject.subscribe(listener) }
+        it 'calls get on the fetcher with version header' do
+          expect_any_instance_of(Fetcher)
+            .to receive(:get)
+            .with(url, headers: { 'Accept' => 'application/json;v=2' })
 
-      it 'emits :cache_miss' do
-        expect(listener).to receive(:cache_miss)
-        performer.call(url).status
+          perform
+        end
       end
 
-      it 'misses on different locale' do
-        expect(listener).to receive(:cache_miss).twice
-        performer.call(url, locale: 'en').status
-        performer.call(url, locale: 'fr').status
-      end
+      context 'with a specific locale' do
+        let(:options) { { locale: 'fr' } }
+
+        it 'calls get on the fetcher with locale header' do
+          expect_any_instance_of(Fetcher)
+            .to receive(:get)
+            .with(url, headers: { 'Accept' => 'application/json', 'Accept-Language' => 'fr' })
 
-      it 'emits :cache_miss' do
-        allow(listener).to receive(:cache_miss)
-        performer.call(url).status
-        expect(listener).to receive(:cache_hit)
-        performer.call(url).status
+          perform
+        end
       end
     end
-  end
 
+    describe '#fget' do
+      let(:options) { {} }
+      let(:perform) { subject.fget(url, **options) }
 
-  describe '#get' do
-    let(:performer) { subject.method(:get) }
+      context 'with no information needed' do
+        it 'does not perform a GET request until we ask for information' do
```

Comment:
It seemed sensible to us to test the negative scenario as well: if a `Future` guarantees that the block will be eventually executed, then I would expect the opposite to be true as well, especially given that the Future is there to avoid making HTTP requests until it is necessary (i.e. for a collection). Is there a specific reason why the opposite wouldn't be true? 

Your second point is sound: the reason why these tests are very unit-y is because we added integration specs that _should_ test the whole thing working together. I can move the `get` examples back into a `shared example` block and call them for both methods.